### ISSUE TITLE
clippy::option-if-let-else suggests a worse code.

### ISSUE THREAD

**Issue by fxdave** [rust-lang/rust-clippy#7006]
Lint name: clippy::option-if-let-else

I tried this code:

```rust
if let Some(draggable) = utils::dom::find(target, "draggable") {
    // some code
} else if let Some(relation) = utils::dom::find(target, "relation") {
    // some code
} else {
    // some code
}
```

I expected to see no errors.

Instead, this happened:

```c
error: use Option::map_or instead of an if let/else
  --> src/components/viewport/mod.rs:78:12
   |
78 |       } else if let Some(relation) = utils::dom::find(target, "relation") {
   |  ____________^
...  |
90 | |     }
   | |_____^
   |
   = note: `-D clippy::option-if-let-else` implied by `-D clippy::pedantic`
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#option_if_let_else
help: try
   |
78 |     } else { utils::dom::find(target, "relation").map_or(
              // some code,
               |relation| {
82 |         // some code
```

### Meta

- `cargo clippy -V`: clippy 0.1.53 (4a20eb6 2021-03-28)
- `rustc -Vv`:
  ```
  rustc 1.53.0-nightly (4a20eb6a9 2021-03-28)
  binary: rustc
  commit-hash: 4a20eb6a9da36c88ee929826c4f1eb8d7ea393b2
  commit-date: 2021-03-28
  host: x86_64-unknown-linux-gnu
  release: 1.53.0-nightly
  LLVM version: 12.0.0
  ```

**Comment by magurotuna** [comment]
So do you mean that the suggested code works fine but feels worse than your original code, right?

**Comment by fxdave** [comment]
Yes

**Comment by camsteffen** [comment]
Yeah...I'd be inclined to just never lint `else if let`.
